controllers/userController.js

```diff
@@ -0,0 +1,80 @@
+const User = require("../models/userModel");
+const catchAsync = require("../utils/catchAsync");
+const jwt = require("jsonwebtoken");
+const AppError = require("../utils/appError");
+const { validationResult } = require("express-validator");
+const { constants } = require("../utils/constants");
+
+const signToken = (id) =>
+  jwt.sign(
+    {
+      id,
+    },
+    process.env.JWT_SECRET,
+    {
+      expiresIn: process.env.JWT_EXPIRES_IN,
+    }
+  );
+
+exports.signup = catchAsync(async (req, res, next) => {
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    return res.status(400).json({ errors: errors.array() });
+  }
+
+  const { email } = req.body;
+
+  const existingUser = await User.findOne({ email });
+
+  if (existingUser) {
+    return next(
+      new AppError(
+        "The email adddress is already in use. Please provide a different one",
+        409
+      )
+    );
+  }
+
+  const newUser = await User.create({
+    name: req.body.name,
+    email: req.body.email,
+    password: req.body.password,
+  });
+  res.status(201).json({
+    status: constants.SUCCESS_TEXT,
+    data: {
+      user: newUser,
+    },
+  });
+});
+
+exports.login = catchAsync(async (req, res, next) => {
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    return res.status(400).json({ errors: errors.array() });
+  }
+  const { email, password } = req.body;
+  // 2. check if email and password is correct
+  const user = await User.findOne({ email }).select("+password");
+
+  if (!user || !(await user.correctPassword(password, user.password))) {
+    return next(new AppError("Incorrect email or password", 401));
+  }
+  // 3. if eveyrthing is ok, send token to client
+  const token = signToken(user._id);
+  res.status(200).json({
+    status: constants.SUCCESS_TEXT,
+    token,
+  });
+});
+
+exports.getAllUsers = catchAsync(async (req, res, next) => {
+  const users = await User.find();
```

Comment:
Not a good idea to return your db model directly. You can always create a separate dto